poetry/installation/installer.py

```diff
@@ -477,7 +478,7 @@ def _execute_uninstall(self, operation: Uninstall) -> None:
         self._installer.remove(operation.package)
 
     def _populate_local_repo(
-        self, local_repo: Repository, ops: List[Operation]
+        self, local_repo: Repository, ops: Sequence[Operation]
```

Comment:
@abn
here's an example of the use of a 'covariant' type (required for *correctness*)

the `_populate_local_repo` method accepts `ops: List[Operation]`, however `Operation` is a base class. In practice we expect that this method should accept a list of some derived class (such as `Uninstall`, for example).

**the problem is that `List[Subclass]` is *not* equivalent to `List[Baseclass]`** (see these docs - http://mypy.readthedocs.io/en/latest/generics.html#variance-of-generic-types)

however `Sequence[Subclass]` *is* equivalent to `Sequence[Baseclass]`. Similarly, `Iterable[Subclass]` *is* equivalent to `Iterable[Baseclass]`.